### ISSUE TITLE
Using objectives reg:squaredlogerror and reg:pseudohubererror does not converge

### ISSUE THREAD

**Issue by honkeat** [dmlc/xgboost#6447]
I am trying to predict y with a range from 0 to 1000000. Using reg:squarederror or reg:linear objective gives an good prediction on the larger values of y, but the prediction on the smaller values of y are bad.

I need a loss that weighs small values and large values equally. I tried to use different objectives reg:squaredlogerror and reg:pseudohubererror but the model doesn't converge at all. 

```import xgboost as xgb
import numpy as np
import random
from sklearn.metrics import explained_variance_score, mean_absolute_error

def MAPE_error(pred, label):
  epsilon = 0.000001
  return ((abs(pred-label)+epsilon)/(label + epsilon)).mean()

# generate data
X = []
y = []

for i in range (10000):
  x1 = random.uniform(0, 100)
  x2 = random.uniform(0, 100)
  x3 = random.uniform(0, 100)

  result = x1 * x2 * x3

  X.append([x1,x2,x3])
  y.append(result) 

reg = xgb.XGBRegressor(objective="reg:linear")
reg.fit(np.array(X), np.array(y))

pred = reg.predict(np.array(X))
print('explained_variance_score:', explained_variance_score(pred, np.array(y)), "mean_absolute_error", mean_absolute_error(pred, np.array(y)), "MAPE", MAPE_error(pred, np.array(y)))
```

Here are the results from the 4 different objectives,
```
linear: 0.9991618961347887 mean_absolute_error 3017.921196522046 MAPE 0.3709115124393537
squarederror: explained_variance_score: 0.9991618961347887 mean_absolute_error 3017.921196522046 MAPE 0.3709115124393537
squaredlogerror: explained_variance_score: -20917431606.197647 mean_absolute_error 125002.23014140659 MAPE 2.0394572901666606
pseudohubererror: explained_variance_score: -57449018161714.95 mean_absolute_error 101892.11100511657 MAPE 58.69028424020796
```

As you can see, the result from using reg:squaredlogerror and reg:pseudohubererror are completely off.

**Comment by trivialfis** [comment]
Any chance you can normalize the value down to a smaller floating-point range?

**Comment by trivialfis** [comment]
I think this is a floating point error issue and can be mitigated by preprocessing.
